go-controller/pkg/util/multi_network.go

```diff
@@ -1383,10 +1389,26 @@ func GetPodNADToNetworkMappingWithActiveNetwork(pod *corev1.Pod, nInfo NetInfo,
 		}
 	}
 
-	networkSelections[activeNetworkNADs[0]] = activeNSE
+	networkSelections[activeNADKey.String()] = activeNSE
 	return true, networkSelections, nil
 }
 
+// getNADWithNamespace returns the first occurrence of NAD key with the given namespace name.
```

Comment:
this comment is a bit funky; can there be multiple instances of a NAD with this NAD key in the same namespace ? 

There's only one, right ?